rclcpp/include/rclcpp/executor.hpp

```diff
@@ -447,6 +403,10 @@ class Executor
     rclcpp::node_interfaces::NodeBaseInterface::SharedPtr node,
     std::chrono::nanoseconds timeout);
 
+  virtual FutureReturnCode spin_until_future_complete_impl(
```

Comment:
We got two colliding styles here, there is also
spin_once -> spin_once_impl and spin_some -> spin_some_impl

Reply:
Added docs

Reply:
> We got two colliding styles here, there is also
> spin_once -> spin_once_impl and spin_some -> spin_some_impl

Yeah, I'll leave it as-is, but I think this is related to the templated functions that take duration vs just taking nanoseconds, I still believe the templated approach is better, and in that case the narrowing to the non-template protected methods are truly just to cast it down to nanoseconds, so that makes sense in those cases. Likely if I had noticed I would have suggesting the newer functions follow that pattern, but it's not that important now.